### DISCUSSION TITLE
API 199 tokens every time

### DISCUSSION THREAD

**Discussion by christiandarkin** [oobabooga/text-generation-webui#1810] [General]
Anyone know why the API returns no more than 199 tokens whatever you set max new tokens to be?
can this limit be raised?

**Comment by LaaZa** [comment]
It should work properly. Are you sure you are passing `max_new_tokens` or `max_length`?

**Comment by christiandarkin** [comment]
here's what i'm passing:
  request = {
        'prompt': modified_prompt,
        'max_new_tokens': 1500,
        'do_sample': True,
        'temperature': .1,
        'top_p': 0.1,
        'typical_p': 1,
        'repetition_penalty': 1.18,
        'top_k': 40,
        'min_length': 0,
        'no_repeat_ngram_size': 0,
        'num_beams': 1,
        'penalty_alpha': 0,
        'length_penalty': 1,
        'early_stopping': False,
        'seed': -1,
        'add_bos_token': True,
        'truncation_length': 2048,
        'ban_eos_token': False,
        'skip_special_tokens': True,
        'stopping_strings': ["#"]

**Reply by christiandarkin** [reply]
it doesn't seem to matter what I put in any of the request settings - apart from "prompt".  it feels like everything else is ignored.

I heard someone say something about using --listen instead of the provided api - but not sure how that works.

**Reply by LaaZa** [reply]
Weird. It works for me as it should. And I didn't see anything wrong in the code.

**Reply by christiandarkin** [reply]
so you're using   HOST = 'localhost:5000'
URI = f'http://{HOST}/api/v1/generate'  - as in the example.py?   and you're managing to get it to listen to the request?  that's funny.  it takes 'prompt' but I don't think it's taking anything else from my request

**Reply by LaaZa** [reply]
I retested and no I don't get this issue. Banning EOS token to generate the full max tokens gives me that for both non-streaming and streaming.
Does it work correctly in the webui? And does setting the setting in the parameters page affect the setting in API?(should not)
What are your settings and the model/type of model(normal, GPTQ, GGML)?

And did you try running the examples?

**Reply by christiandarkin** [reply]
ok - i'm a fool.  I just needed to update - now it works.  thanks!
